src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+        if (!exists()) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.fileDoesNotExist").set("file", path()).handle();
+        }
+
+        return this;
+    }
+
+    /**
+     * Ensures that the given file exists as file
+     *
+     * @return the file itself for fluent method calls
+     * @throws HandledException if the file does not exist or isn't a file
+     */
+    public VirtualFile assertExistingFile() {
+        assertExists();
+
+        if (isDirectory()) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.noFile").set("file", path()).handle();
+        }
+
+        return this;
+    }
+
+    /**
+     * Ensures that the given file exists as directory
+     *
+     * @return the file itself for fluent method calls
+     * @throws HandledException if the file does not exist or isn't a directory
+     */
+    public VirtualFile assertExistingDirectory() {
+        if (!exists()) {
+            throw Exceptions.createHandled()
+                            .withNLSKey("VirtualFile.directoryDoesNotExist")
+                            .set("file", path())
+                            .handle();
+        }
+        if (!isDirectory()) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.noDirectory").set("file", path()).handle();
+        }
+
+        return this;
+    }
+
+    /**
+     * Enumerates all children of this file.
+     *
+     * @param search the search which determines filtering and is responsible for collecting results
+     */
+    public void children(FileSearch search) {
+        try {
+            if (childProvider != null) {
+                childProvider.enumerate(this, search);
+            }
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "childProvider.enumerate");
+        }
+    }
+
+    /**
+     * Tries to find a child with the given name.
+     * <p>
+     * Note that the given file can be non-existent.
+     *
+     * @param name the name of the child to find
+     * @return the child wrapped as optional or an empty optional if the given name cannot be resolved into a file.
+     */
+    public Optional<VirtualFile> findChild(String name) {
+        try {
+            if (childProvider == null) {
+                return Optional.empty();
+            }
+
+            return childProvider.findChild(this, name);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "childProvider.findChild");
+        }
+    }
+
+    /**
+     * Tries to resolve the relative path within this directory.
+     *
+     * @param relativePath the path to resolve
+     * @return the relative path wrapped as optional or an  empty optional if the given relaive path cannot be
+     * resolved into a file.
+     */
+    public Optional<VirtualFile> tryResolve(String relativePath) {
+        if (relativePath != null && relativePath.startsWith("/")) {
+            relativePath = relativePath.substring(1);
```

Comment:
Extract method maybe? Altering a parameter is a code smell (doesn't SonarQube warn about this? It did before)